src/ILLink.RoslynAnalyzer/IMethodSymbolExtensions.cs

```diff
@@ -0,0 +1,85 @@
+// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using ILLink.Shared.TypeSystemProxy;
+using Microsoft.CodeAnalysis;
+
+namespace ILLink.RoslynAnalyzer
+{
+	internal static class IMethodSymbolExtensions
+	{
+		public static bool HasImplicitThis (this IMethodSymbol method)
+		{
+			return !method.IsStatic;
+		}
+
+		/// <summary>
+		/// Returns a list of the parameters pushed onto the stack before the method call (including the implicit 'this' parameter)
+		/// </summary>
+		public static List<ParameterProxy> GetParameters (this IMethodSymbol method)
+		{
+			List<ParameterProxy> parameters = new ();
```

Comment:
I agree, I don't love the idea of allocating for each call. I'm not 100% sure how foreach works with IEnumerable, but we could make a struct that implements IEnumerable<ParamterProxy> and return that struct from GetParameters(), but I'm guessing it would get boxed by the foreach. We'd be allocating the IEnumerable and IEnumerator instead of a whole list which could have some gains, but we'd lose the foreach optimization for List.

Reply:
Created a `struct ParameterCollection` to avoid allocating.